stac_fastapi/types/setup.py

```diff
@@ -9,7 +9,7 @@
     "fastapi>=0.109.0",
     "attrs>=23.2.0",
     "pydantic-settings>=2",
-    "stac_pydantic~=3.1",
+    "stac_pydantic>=3.1.3,<4.0",
```

Comment:
this is to avoid having the `stac.Collections` typedDict different than the stac-pydantic model